client/src/modules/vouchers/voucher-registry.ctrl.js

```diff
@@ -116,6 +116,9 @@ function VoucherController(
   function search() {
     var filtersSnapshot = Vouchers.filters.formatHTTP();
 
+    // Keep the last filters made during the last search and keeps the display in a readable format
```

Comment:
This is an interesting way to pass filters in ... but it would probably be better to use the service to get the filters in the modal controller.  It should be as simple as importing `VoucherService` into the `ModalController` and then doing `var filters = VoucherService.filters.formatView();`

Reply:
Having read all the code now, I actually don't think you need to share the view values at all.  It should all be driven by the search modal.